teps/0091-trusted-resources.md

```diff
@@ -313,6 +342,104 @@ What other approaches did you consider and why did you rule them out?  These do
 not need to be as detailed as the proposal, but should include enough
 information to express the idea and why it was not acceptable.
 -->
+### Options to mitigate the possible impact from mutating webhook
+Some possible mitigations/solutions include the following:
+
+1. Verify the Tasks/Pipelines when applying them in the cluster, and update the Resources to add the missing mutating values (e.g. Use `SetDefaults`) everytime we bump the version of Tekton Pipeline.
+2. Avoid mutating Tekton Resources or move the mutating after verification.
+3. Store the encoded resource in annotation before mutating and verify the stored resource at validating webhook.
+4. Do verification in mutating webhook, if failed verification then mark the run as failed and later check and reject the resource in validationg webhook.
+
+**Pros vs Cons:**
+| Option | Pros | Cons |
+| -------- | ----------- | ----------- |
+| Call SetDefaults() when signing | Help users add missing values |  Need to update signature when there is update in mutating
+| Skip mutating webhook | Address the mutating issue, can also help to detect malicious mutating webhook| Extra cost for Tekton Pipeline developers
+| Store the object at mutating webhook before mutating| Fix the issue without skipping the mutating webhook |  May not work if the attck happens in mutating webhook and only modify the spec
+| Verify at mutating webhook | No need to skip the mutating or store in annotation | Doesn't work if there is a malicious mutating webhook after the normal mutating webhook
+
+In this TEP we propose to proceed with option 1 and together with option 2.
+
+Option 1 can help to add missing values when users creating resources yaml files.
```

Comment:
I think this is probably more trouble than it's worth.

In order for this to work, client versions need to stay in sync with the server (and potentially also other server-side configuration like the alpha feature flag). Any time either of these change, you risk breaking the signatures which would be a backwards incompatible change, which can also have rippling effects for the catalog, etc.

If we're skipping the mutating webhook, this already means that we're not modifying the object from what the user already gave us - instead of requiring a replay of the defaulting behavior we can just validate the message as the user gave it to us, and handling the defaulting at reconcile.

Reply:
Sure that makes sense! I will remove this.